test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1112 @@
+        type  select: string.
+
+        param ExpectSuccess: Whether the caller expects the submission to be
+                             successful or not.
+        type  ExpectSuccess: int.
+        """
+        start = int(time.time()) + offset
+        end = start + duration
+
+        if standing:
+            attrs = {'Resource_List.select': select,
+                     'reserve_start': start,
+                     'reserve_end': end,
+                     'reserve_timezone': self.tzone,
+                     'reserve_rrule': rrule}
+        else:
+            attrs = {'Resource_List.select': select,
+                     'reserve_start': start,
+                     'reserve_end': end}
+
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+        msg = "Resv;" + rid + ";New reservation submitted start="
+        msg += time.strftime(self.fmt, time.localtime(int(start)))
+        msg += " end="
+        msg += time.strftime(self.fmt, time.localtime(int(end)))
+        if standing:
+            msg += " recurrence_rrule=" + rrule + " timezone="
+            msg += self.tzone
+
+        self.server.log_match(msg, interval=2, max_attempts=30)
+
+        if ExpectSuccess:
+            attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            self.server.expect(RESV, attrs, id=rid)
+
+            msg = "Resv;" + rid + ";Reservation confirmed"
+            self.server.log_match(msg, interval=2,
+                                  max_attempts=30)
+
+            self.server.expect(RESV, attrs, id=rid)
+            acct_msg = "U;" + rid + ";requestor=" + TEST_USER.name + "@"
+            acct_msg += self.server.shortname
+
+            if standing:
+                acct_msg += " recurrence_rrule=" + re.escape(rrule)
+                acct_msg += " timezone=" + re.escape(self.tzone)
+
+            self.server.accounting_match(acct_msg, interval=2, regexp=True,
+                                         max_attempts=30)
+        else:
+            msg = "Resv;" + rid + ";Reservation denied"
+            self.server.log_match(msg, interval=2,
+                                  max_attempts=30)
+
+        return rid, start, end
+
+    def check_resv_running(self, rid, offset=0, display=True):
+        """
+        Helper method to wait for reservation to start running.
+
+        param rid: Reservation id.
+        type  rid: string.
+
+        param offset: Time in seconds to wait for the reservation to start
+                      running.
+        type  offset: int.
+
+        param display: Whether to display a message or not, default - Yes.
+        type  display: bool
+        """
+        if offset > 0:
+            if display:
+                self.logger.info("Waiting for reservation to start running.")
+            else:
+                self.logger.info("Sleeping.")
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, attrs, id=rid, offset=offset)
```

Comment:
expect is iterative. Iterations can be overwritten. Fixed waiting is not desirable.

Reply:
If that is the case, why does the expect method provides an offset option. I will keep the expect as it is.